### PR TITLE
Core/AI: Factory functions cleanup 

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes 

## SOURCE:

https://github.com/TrinityCore/TrinityCore/commit/ce3787f190d28c1e981598eef2cd24c622a2cbae
## Tests Performed:

- Part of dynamic go/create & Smart_scripts
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Nefertumm — APPROVED** [review]

**Comment by acidmanifesto** [comment]
How to test? or is this all code style changes? i'm not really unstanding this.

**Comment by acidmanifesto** [comment]
Merge Conflicts

**Comment by acidmanifesto** [comment]
What is the status of this PR?

**Comment by FrancescoBorzi** [comment]
good job @IntelligentQuantum and thanks @Nefertumm and @Maelthyrr for reviewing/testing this.

File: src/server/game/AI/CreatureAISelector.cpp [hunk 1]
```diff
@@ -15,132 +15,102 @@
...
 
-    MovementGenerator* selectMovementGenerator(Creature* creature)
+    MovementGenerator* SelectMovementGenerator(Unit* unit)
     {
-        MovementGeneratorRegistry& mv_registry(*MovementGeneratorRegistry::instance());
-        ASSERT(creature->GetCreatureTemplate());
-        const MovementGeneratorCreator* mv_factory = mv_registry.GetRegistryItem(creature->GetDefaultMovementType());
+        MovementGeneratorType type = IDLE_MOTION_TYPE;
+        if (unit->GetTypeId() == TYPEID_UNIT)
+            type = unit->ToCreature()->GetDefaultMovementType();
```
Comment (pklloveyou) [inline src/server/game/AI/CreatureAISelector.cpp:83]:
```
        if (unit->GetTypeId() == TYPEID_UNIT)
        {
            type = unit->ToCreature()->GetDefaultMovementType();
            if (type == FLEEING_MOTION_TYPE)
                type = IDLE_MOTION_TYPE;
        }
```
Prevent service crash

**Pull request merged**